bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/client/MagentoGraphqlClientImpl.java

```diff
@@ -190,10 +191,13 @@ private void initModel(Resource resource, Page page, SlingHttpServletRequest req
         }
 
         this.httpHeaders = headers;
+        // In certain situations resource.getResourceType() returns an enforced resource type.
+        // We prefer the resource type of the component proxy for the cache name.
+        String cacheName = resource.getValueMap().get(ResourceResolver.PROPERTY_RESOURCE_TYPE, resource.getResourceType());
```

Comment:
This may be null, please fallback to `resource.getResourceType()` if so

Reply:
What may be null here?

Reply:
ahh mist the default, nevermind and sorry.